modules/@apostrophecms/util/ui/public/utils.js

```diff
@@ -168,29 +168,16 @@
   // This is a computer science principle known as "separation of concerns."
 
   apos.util.runPlayers = function(el) {
-    var widgets = (el || document).querySelectorAll('[data-apos-widget]');
-    var i;
-    if (el && el.getAttribute('data-apos-widget')) {
-      // el is itself a widget. Might still contain some too
-      play(el);
-    }
-    for (i = 0; (i < widgets.length); i++) {
-      play(widgets[i]);
-    }
+    const players = apos.util.widgetPlayers;
+    const playerList = Object.keys(players);
 
-    function play(widget) {
-      if (widget.getAttribute('data-apos-played')) {
-        return;
-      }
-      var data = JSON.parse(widget.getAttribute('data'));
-      var options = JSON.parse(widget.getAttribute('data-options'));
-      widget.setAttribute('data-apos-played', '1');
-      // bc with the old lean module
-      var player = apos.util.widgetPlayers[data.type] || (apos.lean && apos.lean.widgetPlayers && apos.lean.widgetPlayers[data.type]);
-      if (!player) {
-        return;
-      }
-      player(widget, data, options);
+    for (let i = 0; i < playerList.length; i++) {
+      const playerOpts = players[playerList[i]];
+      const playerEls = (el || document).querySelectorAll(playerOpts.selector);
+
+      playerEls.forEach(function (el) {
+        playerOpts.player(el);
```

Comment:
We still want to add data-apos-played and not play them if we have already played them. Yes this is a data attribute of ours on the front end but nobody has to use players if they don't want to.